Plug-Ins/NovaTools/System Editor/StellarView.swift

```diff
@@ -0,0 +1,210 @@
+import RFSupport
+
+class StellarView: NSView {
+    let resource: Resource
+    let manager: RFEditorManager
+    let isEnabled: Bool
+    private(set) var position = NSPoint.zero
+    private(set) var image: NSImage? = nil
+    private var attributes: [NSAttributedString.Key: Any] {
+        [.foregroundColor: NSColor.white, .font: NSFont.systemFont(ofSize: 11)]
+    }
+    private var systemView: SystemMapView? {
+        superview as? SystemMapView
+    }
+    var point = NSPoint.zero {
+        didSet {
+            frame.origin.x = point.x - frame.size.width / 2
+            frame.origin.y = point.y - frame.size.height / 2
+        }
+    }
+    var isHighlighted = false {
+        didSet {
+            if oldValue != isHighlighted {
+                needsDisplay = true
+            }
+        }
+    }
+    var highlightCount = 1 {
+        didSet {
+            if oldValue != highlightCount {
+                needsDisplay = true
+            }
+        }
+    }
+
+    init?(_ stellar: Resource, manager: RFEditorManager, isEnabled: Bool) {
+        self.resource = stellar
+        self.manager = manager
+        self.isEnabled = isEnabled
+        super.init(frame: .zero)
+        do {
+            try self.read()
+        } catch {
+            return nil
+        }
+    }
+
+    func read() throws {
+        let reader = BinaryDataReader(resource.data)
+        position.x = Double(try reader.read() as Int16)
+        position.y = Double(try reader.read() as Int16)
+
+        let spinId = (Int)(try reader.read() as Int16) + 1000;
+        if let spinResource = manager.findResource(type: ResourceType("spïn"), id: spinId, currentDocumentOnly: false) {
+            let spinReader = BinaryDataReader(spinResource.data)
+            let spriteId = (Int)(try spinReader.read() as Int16);
+
+            // Use the preview provider to obtain an NSImage from the resources the spïn might point to -- try PICT then fall back to rlëD
+            if let spriteResource = manager.findResource(type: ResourceType("PICT"), id: spriteId, currentDocumentOnly: false) ?? manager.findResource(type: ResourceType("rlëD"), id: spriteId, currentDocumentOnly: false) {
+                spriteResource.preview({ img in
+                    self.image = img
+                    self.updateFrame()
+                    self.needsDisplay = true
+                })
+            }
+        }
+    }
+
+    override func updateTrackingAreas() {
+        for trackingArea in self.trackingAreas {
+            self.removeTrackingArea(trackingArea)
+        }
+        let tracking = NSTrackingArea(rect: bounds, options: [.mouseEnteredAndExited, .activeInKeyWindow, .enabledDuringMouseDrag], owner: self, userInfo: nil)
+        self.addTrackingArea(tracking)
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    func updateFrame() {
+        if image != nil, let imageSize = systemView?.transform.transform(image!.size) {
```

Comment:
Prefer using `if let` to unwrap here.
```suggestion
        if let image, let imageSize = systemView?.transform.transform(image.size) {
```